File: examples/function.zig
const napi = @import("napi");
const std = @import("std");

pub fn callAddTwoNumbers(env: napi.Env, num: napi.Value) !napi.Value {
    const global = try env.getGlobal();
    const add_two = try global.getNamedProperty("AddTwo");
    const args = [_]napi.Value{num};
    return add_two.callFunction(args.len, global, args);
}

comptime {
    napi.registerModule(init);
}

fn init(env: napi.Env, exports: napi.Value) !napi.Value {
    try exports.setNamedProperty(
        "callAddTwoNumbers",
        try env.createFunction(callAddTwoNumbers, null),
    );

    return exports;
}

File: examples/function.js
const addon = require('../zig-out/lib/function.node');
const assert = require('node:assert/strict');

function AddTwo(num) {
  return num + 2;
}
global.AddTwo = AddTwo;

assert.strictEqual(3, addon.callAddTwoNumbers(1));
assert.strictEqual(103, addon.callAddTwoNumbers(101));
